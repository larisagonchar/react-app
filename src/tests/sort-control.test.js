import { render, screen } from '@testing-library/react';
import SortControl from '../components/sort-control/sort-control';
import userEvent from '@testing-library/user-event';

describe('SortControl component', () => {  
  test('render sort control', () => {
    const { asFragment } = render(<SortControl selectedControl={'Title'} />);

    expect(asFragment).toMatchSnapshot();
  });

  test('call onSelect when user click on item', () => {
    const onItemClickMock = jest.fn();
    const option = 'Release Date';
    render(<SortControl selectedControl={'Title'} onSelect={onItemClickMock}/>);

    const select = screen.getByTestId('select');

    userEvent.selectOptions(select, option);

    expect(onItemClickMock).toHaveBeenCalledWith(option);
  });
});