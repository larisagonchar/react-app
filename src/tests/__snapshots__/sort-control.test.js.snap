// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortControl component render sort control 1`] = `[Function]`;
